src/Take.Blip.Builder/Utils/SalesForce/SalesForceClient.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Models;
+using Take.Blip.Builder.Utils.SalesForce.Models;
+
+namespace Take.Blip.Builder.Utils.SalesForce
+{
+    public class SalesForceClient : ISalesForceClient
+    {
+        private readonly ILogger _logger;
+        private const string LEAD_ATTRIBUTE_KEY = "attributes";
+
+        public SalesForceClient(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task<AuthorizationResponse> GetAuthorizationAsync(SalesForceConfig salesForceConfig, string ownerId, CancellationToken cancellationToken)
+        {
+            var body = new Dictionary<string, string>()
+            {
+                {"grant_type", "hybrid_refresh" },
+                {SalesForceConstants.CLIENT_ID, salesForceConfig.ClientId },
+                {SalesForceConstants.CLIENT_SECRET, salesForceConfig.ClientSecret},
+                {SalesForceConstants.REFRESH_TOKEN, salesForceConfig.RefreshToken }
+            };
+
+            var responseToken = string.Empty;
+
+            try
+            {
+                using (HttpClient client = new HttpClient())
+                {
+                    HttpResponseMessage response = client.PostAsync(
+                        $"{SalesForceConstants.SALES_FORCE_URI}{SalesForceRoutes.REFRESH_TOKEN}",
+                        new FormUrlEncodedContent(body)
+                        ).Result;
+                    responseToken = await response.Content.ReadAsStringAsync();
+                };
+                return JsonConvert.DeserializeObject<AuthorizationResponse>(responseToken);
+            }
+            catch (Exception ex)
+            {
+                _logger.Warning(ex, $"An exception occurred while processing a Sales Force  HTTP request");
```

Comment:
Its a warning? The user will continue to use the functionality whitout crashing the flow?